Kernel/Net/IPv4Socket.cpp

```diff
@@ -637,7 +637,7 @@ ErrorOr<void> IPv4Socket::ioctl(OpenFileDescription&, unsigned request, Userspac
             if (route.rt_gateway.sa_family != AF_INET)
                 return EAFNOSUPPORT;
             if (!(route.rt_flags & RTF_UP))
-                return EINVAL; // FIXME: Find the correct value to return
+                return EEXIST;
```

Comment:
The `route` struct comes from userspace (see line 623) so how is this a check for duplicate routes? I think what this error needs to convey, is that you can't add a route if you made it unusable by not setting `RTF_UP`.

Reply:
I got the bitwise compare backwards in my head and went with it because this section doesn't make sense otherwise. To be clear, the if will pass when rt_flags does not contain RTF_UP, correct? If so, that means there are 9 other flags that could be present and a single error doesn't cover all of them. Either the ! needs to come out, or more errors need to go in. Let me know if I'm on the right track and I'll sort this tomorrow.

Reply:
Well, I'm not really sure what your end goal was with this PR?

1) If you want to get rid of the `FIXME`, then you need to verify that `EINVAL` is indeed the best error to return there. The code is currently checking if the user passed a route struct without `RTF_UP` being set, which would mean that the route was deleted and doesn't make any sense in the context of adding a new route.

2) If you want to return another error when a route already exists (duplicate route), then you need to find the code that checks for that. These checks only look at the user-provided route struct, so we can't trust what's in there.